### PR TITLE
Don't treat template generated files with assembly attributes as gene…

### LINKED ISSUE #4390: AssemblyInfo.cs is not a generated file
The batch fixer for our code fixes is not applying changes to **AssemblyInfo.cs** because it thinks it is a generated file. This a problem for two reasons:

1. The file [is not a generated file](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/StyleCop.Analyzers/StyleCopTester/Properties/AssemblyInfo.cs). It is a normal file created by most of the project templates in Visual Studio.
2. It should not be the job of the `FixAllProvider` to separate generated files from non-generated files. The `DiagnosticAnalyzer` and/or `CodeFixProvider` should bear the responsibility for suppressing messages in generated files. By placing this code in the `FixAllProvider`, the UI indicates a problem was found and an automatic solution is available, but then fails to apply the automatic solution. See #2753 for more information on this.

### PR DESCRIPTION
**Pull request by mavasani**
…rated code (e.g. AssemblyInfo.cs)

Fixes #4390

### CODE REVIEW

**Comment by mavasani** [comment]
@dotnet-bot  retest win please

**Comment by mavasani** [comment]
@heejaechang @basoundr @jmarolf  can you please review, should be a quick review.

File: src/Workspaces/Core/Portable/GeneratedCodeRecognition/GeneratedCodeRecognitionServiceFactory.cs [hunk 1]
```diff
@@ -37,12 +37,10 @@ private static bool IsFileNameForGeneratedCode(string fileName)
                 {
                     fileName = Path.GetFileNameWithoutExtension(fileName);
 
-                    if (fileName.EndsWith("AssemblyInfo", StringComparison.OrdinalIgnoreCase) ||
-                        fileName.EndsWith(".designer", StringComparison.OrdinalIgnoreCase) ||
+                    if (fileName.EndsWith(".designer", StringComparison.OrdinalIgnoreCase) ||
```
Comment (heejaechang) [inline src/Workspaces/Core/Portable/GeneratedCodeRecognition/GeneratedCodeRecognitionServiceFactory.cs]: isn't this generic service? is that okay to change default to meet suppression's need? how about other consumers?

**Comment by heejaechang** [comment]
:+1:

File: src/Workspaces/Core/Portable/GeneratedCodeRecognition/GeneratedCodeRecognitionServiceFactory.cs (hunk 1 above)
Comment (mavasani) [inline src/Workspaces/Core/Portable/GeneratedCodeRecognition/GeneratedCodeRecognitionServiceFactory.cs]: @heejaechang This change is not needed for suppression, it is an unrelated bug fix. This service indicates whether a source file is a temporary generated file, that is likely to change/regenerate during build or in the designer. AssemblyInfo files are not in this category, it was decided in triage that files generated by project templates shouldn't be considered as temporary generated files.

**Pull request merged**